### ISSUE TITLE
Kubeflow on-prem deployment  Jupyter notebooks not working

### ISSUE THREAD

**Issue by Tabrizian** [kubeflow/kubeflow#3222]
Using  a fresh installation of Kubeflow on Kubernetes, it seems like ```notebook-controller``` is unable to assign correct serviceaccount to Pods created for notebooks. Here is the logs for the stateful app:

```
"my-kubeflow-0" is forbidden: error looking up service account kubeflow/jupyter-notebook: serviceaccount "jupyter-notebook" not foun
```

**Comment by LouNauSRC** [comment]
Having the same issue spinning up a notebook server using version 0.50. 
Version 0.51 does not appear to have a kfctl tar for linux, just darwin.
Environment is a 3 node K8s cluster stood up and controlled by Rancher.

I had previously had Kubeflow 0.10 installed via the Rancher helm chart, but removed that before installing 0.50 via kfctl.

**Comment by solovyevt** [comment]
Just tested it on a fresh Rancher 2 k8s deployment, everything works fine.

Service account jupyter-notebook is being created by ksonnet during `kfctl apply -V` deployment. Do you have any error log messages during the deployment process?

**Comment by LouNauSRC** [comment]
See attached file for the kfctl generate, and the kfctl apply.  No errors that I see.  And no mention of jupyter-notebook.

[kfctl_output.txt](https://github.com/kubeflow/kubeflow/files/3162837/kfctl_output.txt)

**Comment by solovyevt** [comment]
Okay, I reproduced that with fresh 0.5.0 deployment.

Jupyter notebook StatefulSet events:
`Events:
  Type     Reason        Age                 From                    Message
  ----     ------        ----                ----                    -------
  Warning  FailedCreate  25s (x15 over 64s)  statefulset-controller  create Pod test-0 in StatefulSet test failed error: pods "test-0" is forbidden: error looking up service account kubeflow-workspace/jupyter-notebook: serviceaccount "jupyter-notebook" not found`

**Comment by solovyevt** [comment]
Update: just read `kfctl init --help` and found out an interesting fact:

`  -v, --version string          desired version of Kubeflow or master if not specified. Version can be master (eg --version master) or a git tag (eg --version=v0.5.0), or a PR (eg --version pull/<id>). (default "master")`

So I've tried deploying kubeflow 0.5.0 with `kfctl init --version=v0.5.0 $KFAPP` and jupyter-notebook service account was successfully created.

**Comment by jlewi** [comment]
I believe this should be fixed in v0.5.1 by #3205
